models/eprop_iaf_psc_delta.cpp

```diff
@@ -313,18 +313,18 @@ eprop_iaf_psc_delta::update( Time const& origin, const long from, const long to
   {
     const long t = origin.get_steps() + lag;
 
-    const auto z_in = B_.spikes_.get_value( lag );
+    auto z_in = B_.spikes_.get_value( lag );
 
     if ( S_.r_ == 0 ) // not refractory, can spike
     {
-      S_.v_m_ = V_.P_i_in_ * ( S_.i_in_ + P_.I_e_ ) + V_.P_v_m_ * S_.v_m_ + z_in;
-
       if ( P_.with_refr_input_ and S_.refr_spikes_buffer_ != 0.0 )
       {
-        S_.v_m_ += S_.refr_spikes_buffer_;
+        z_in += S_.refr_spikes_buffer_;
         S_.refr_spikes_buffer_ = 0.0;
       }
 
+      S_.v_m_ = V_.P_i_in_ * ( S_.i_in_ + P_.I_e_ ) + V_.P_v_m_ * S_.v_m_ + z_in;
+
```

Comment:
For me, the original version is easier to follow because the propagator is applied directly to $v_m$, and the line `S_.v_m_ += S_.refr_spikes_buffer_`  indicates a direct modification to $v_m$. However, this might just come down to personal preference.

Reply:
That's a good observation! I agree that having all the lines modifying `v_m` visible at a glance is crucial, so I’ve reverted that commit.